bundles/WebToPrintBundle/src/Processor/Chromium.php

```diff
@@ -0,0 +1,159 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\WebToPrintBundle\Processor;
+
+use HeadlessChromium\BrowserFactory;
+use Pimcore\Bundle\WebToPrintBundle\Config;
+use Pimcore\Bundle\WebToPrintBundle\Event\Model\PrintConfigEvent;
+use Pimcore\Bundle\WebToPrintBundle\Model\Document\PrintAbstract;
+use Pimcore\Bundle\WebToPrintBundle\Processor;
+use Pimcore\Bundle\WebToPrintBundle\Event\DocumentEvents;
+use Pimcore\Logger;
+use Pimcore\Image\Chromium as ChromiumLib;
+
+class Chromium extends Processor
+{
+    /**
+     * @internal
+     */
+    protected function buildPdf(PrintAbstract $document, object $config): string
+    {
+        $web2printConfig = Config::getWeb2PrintConfig();
+        $web2printConfig = $web2printConfig['chromiumSettings'];
+        $web2printConfig = json_decode($web2printConfig, true);
+
+        $params = [
+            'document' => $document,
+            'hostUrl' => 'http://nginx:80'
+        ];
+        $this->updateStatus($document->getId(), 10, 'start_html_rendering');
+        $html = $document->renderDocument($params);
+
+        if (isset($web2printConfig['hostUrl'])) {
+            $params['hostUrl'] = $web2printConfig['hostUrl'];
+        }
+
+        $html = $this->processHtml($html, $params);
+        $this->updateStatus($document->getId(), 40, 'finished_html_rendering');
+
+        if ($web2printConfig) {
+            foreach (['header', 'footer'] as $item) {
+                if (key_exists($item, $web2printConfig) && $web2printConfig[$item] &&
+                    $content = file_get_contents($web2printConfig[$item])) {
+                    $web2printConfig[$item . 'Template'] = $content;
+                }
+                unset($web2printConfig[$item]);
+            }
+        }
+
+        try {
+            $this->updateStatus($document->getId(), 50, 'pdf_conversion');
+            $pdf = $this->getPdfFromString($html, $web2printConfig);
```

Comment:
I tried to render my web2print test like for gotenberg but I do get the following error in `var/log/php.log`

```php
[2023-02-27T13:08:45.702361+00:00] messenger.CRITICAL: Error thrown while handling message Pimcore\Bundle\WebToPrintBundle\Messenger\GenerateWeb2PrintPdfMessage. Removing from transport after 3 retries. Error: "Handling "Pimcore\Bundle\WebToPrintBundle\Messenger\GenerateWeb2PrintPdfMessage" failed: Pimcore\Bundle\WebToPrintBundle\Processor\Chromium::getPdfFromString(): Argument #2 ($params) must be of type array, null given, called in /var/www/html/dev/pimcore/pimcore/bundles/WebToPrintBundle/src/Processor/Chromium.php on line 65" {"class":"Pimcore\\Bundle\\WebToPrintBundle\\Messenger\\GenerateWeb2PrintPdfMessage","retryCount":3,"error":"Handling \"Pimcore\\Bundle\\WebToPrintBundle\\Messenger\\GenerateWeb2PrintPdfMessage\" failed: Pimcore\\Bundle\\WebToPrintBundle\\Processor\\Chromium::getPdfFromString(): Argument #2 ($params) must be of type array, null given, called in /var/www/html/dev/pimcore/pimcore/bundles/WebToPrintBundle/src/Processor/Chromium.php on line 65","exception":"[object] (Symfony\\Component\\Messenger\\Exception\\HandlerFailedException(code: 0): Handling \"Pimcore\\Bundle\\WebToPrintBundle\\Messenger\\GenerateWeb2PrintPdfMessage\" failed: Pimcore\\Bundle\\WebToPrintBundle\\Processor\\Chromium::getPdfFromString(): Argument #2 ($params) must be of type array, null given, called in /var/www/html/dev/pimcore/pimcore/bundles/WebToPrintBundle/src/Processor/Chromium.php on line 65 at /var/www/html/vendor/symfony/messenger/Middleware/HandleMessageMiddleware.php:129)\n[previous exception] [object] (TypeError(code: 0): Pimcore\\Bundle\\WebToPrintBundle\\Processor\\Chromium::getPdfFromString(): Argument #2 ($params) must be of type array, null given, called in /var/www/html/dev/pimcore/pimcore/bundles/WebToPrintBundle/src/Processor/Chromium.php on line 65 at /var/www/html/dev/pimcore/pimcore/bundles/WebToPrintBundle/src/Processor/Chromium.php:95)"} []
```